### DISCUSSION TITLE
Help with log file interpretation.

### DISCUSSION THREAD

**Discussion by jppr12** [shadow/shadow#740] [General]
I'm trying to have a full understand of the messages present in the log file generate by shadow, and i'm not 100% sure of what is the difference between the:
inbound-localhost-counters, outbound-localhost-counters, inbound-remote-counters and the outbound-remote-counters.

I would be very grateful if you could help me with this issue.
By the way, and a small aside, there are any new examples of simple simulations compatible with the updates made in tgen?

**Comment by robgjansen** [comment]
Inbound/outbound tracks packets, bytes, etc. received/sent on a particular network interface. Localhost are for connections over localhost, whereas remote are for connections to a host on a different machine.

The inbound/outbound counters that are printed for localhost are also printed for remote connections. So there is a lot of data present in the log file. In case you want to read the code that formats the string that is printed to the log file, that is here:

https://github.com/shadow/shadow/blob/d54fa6321b293ac03856777cb805990691d9149e/src/main/host/shd-tracker.c#L385-L403

The new TGen that we just released has examples of running in Shadow here:
https://github.com/shadow/tgen/tree/master/resource

**Comment by jppr12** [comment]
Hi, thanks for the help. I'm not yet understanding 100% how this works but my problem now is with the relation between what is configured in the shadow.config.xml and the results of the simulation.

So, in the example resource/examples/shadow.config.xml, for what i understand, we have 2 servers and 1 client, so i can't understand how that hosts are related with what is configured in the tophology element. Who is id="poi-1" and why are both source and target on the edge part "poi-1".

The other question, executing the example on that folder, is why the log file only have representative values on the remote side of the counter? i suppose the connection is directed from client to servers, why in any point the are no values on the localhost side? When a client recieve packets destined for him the values shoulden't be on that side?

Thanks for the attention.

**Comment by robgjansen** [comment]
> So, in the example resource/examples/shadow.config.xml, for what i understand, we have 2 servers and 1 client, so i can't understand how that hosts are related with what is configured in the tophology element. Who is id="poi-1" and why are both source and target on the edge part "poi-1".

The topology section represents the Internet map. Think of "poi-1" as the Internet service provider, or the router that hosts can get attached to. The edge between "poi-1" and "poi-1" shows the properties that will be enforced when any two hosts (client and servers) that are attached to "poi-1" attempt to communicate with each other. Since "poi-1" is the only "router", all hosts will get attached to it and have e.g. 50 milliseconds of latency between them. (Probably a better name for "poi-1" is "isp".)

> The other question, executing the example on that folder, is why the log file only have representative values on the remote side of the counter? i suppose the connection is directed from client to servers, why in any point the are no values on the localhost side? When a client recieve packets destined for him the values shoulden't be on that side?

"localhost" and "remote" do not identify the *endpoints* of a particular connection, but rather it tells you if the connection is on "localhost" connecting to another service on the same host, vs. a "remote" connection that is connecting to a server on a different host. In the example, the client and servers are not run on the same machine, so all connections are "remote" and there are no "localhost" connections.
